### PR TITLE
Allow building Kubernetes in parallel

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This speeds up a run of `krel stage` by ~10 minutes.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
/hold
Requires https://github.com/kubernetes/kubernetes/pull/96882
#### Does this PR introduce a user-facing change?

```release-note
Enable parallel Kubernetes build on `krel stage` 
```

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
/lgtm

**Review by xmudrii — APPROVED** [review]
/lgtm

**Comment by puerco** [comment]
@saschagrunert if we run the build process in a single machine we are still only getting 28GB RAM top. Is it enough?

I've been playing with the idea of launching the builds in parallel, but using several GCB steps which would give each build process it's own VM. Do you think it is still worth pursuing that idea?

**Comment by saschagrunert** [comment]
> @saschagrunert if we run the build process in a single machine we are still only getting 28GB RAM top. Is it enough?

It's enough to speed-up the build. I don't think that we will encounter OOM kills, but maybe the machine has to swap a bit if swap is configured.

> I've been playing with the idea of launching the builds in parallel, but using several GCB steps which would give each build process it's own VM. Do you think it is still worth pursuing that idea?

Hardware wise, I think the whole build runs on 1 virtual machine whereas we can specify the steps in container images. This means we would not safe resources by splitting up the build into dedicated steps. We would have to create dedicated build jobs to achieve the goal, but this would increase the complexity of a `stage` run.

**Comment by justaugustus** [comment]
@saschagrunert -- Is this one ready to go?

**Comment by xmudrii** [comment]
@justaugustus -- AFAIK, this depends on https://github.com/kubernetes/kubernetes/pull/96882 which is still not merged.

**Comment by saschagrunert** [comment]
Yep, pinged some approvers again on kubernetes/kubernetes#96882

**Comment by puerco** [comment]
https://github.com/kubernetes/kubernetes/pull/96882 has merged, we can now go forward with this one !

/retest

**Comment by xmudrii** [comment]
Seems like the error is related to the permissions. Maybe it's just temporary/flake.
/retest

**Comment by xmudrii** [comment]
@saschagrunert Is it okay to remove hold from this PR?

**Comment by saschagrunert** [comment]
> @saschagrunert Is it okay to remove hold from this PR?

Nope, l was going to do some manual test again but it looks like we cannot build releases right now: 
https://console.cloud.google.com/cloud-build/builds/c3e45665-b8d3-4e85-b3bd-49ef31ef011a;step=2?project=kubernetes-release-test

**Comment by saschagrunert** [comment]
We save ~10 minutes with this change. 
/hold cancel

**Review by xmudrii — APPROVED** [review]
/lgtm

**Comment by puerco** [comment]
Awesome @saschagrunert  !!

**Comment by justaugustus** [comment]
Great work!!

**Pull request merged**
